src/launcher.rs

```diff
@@ -881,4 +925,44 @@ mod tests {
         let error = Launcher::new(Some(V2.clone())).unwrap_err().to_string();
         assert_eq!(error, "the requested version (2.0.0) is not installed");
     }
+
+    #[test]
+    fn should_run_existing_shutdown_script() {
+        let _ = logging::init();
+
+        let script_path = Path::new(SHUTDOWN_SCRIPT_PATH);
+        let output_path = Path::new(SHUTDOWN_SCRIPT_OUTPUT_PATH);
+        if script_path.exists() {
+            fs::remove_file(script_path).expect("Couldn't remove existing test shutdown script");
+        }
+        if output_path.exists() {
+            fs::remove_file(output_path)
+                .expect("Couldn't remove existing test shutdown script output");
+        }
+        let script_contents = format!(
+            "#!/bin/bash\necho \"{}\" > {} && exit {}",
+            SHUTDOWN_SCRIPT_SUCCESS_OUTPUT, SHUTDOWN_SCRIPT_OUTPUT_PATH, SHUTDOWN_SCRIPT_EXIT_CODE
+        );
+        fs::write(script_path, script_contents).expect("Couldn't write shutdown script contents");
+        let mut script_perms = fs::metadata(script_path)
+            .expect("Couldn't read shutdown script permissions")
+            .permissions();
+        script_perms.set_mode(0o744);
+        fs::set_permissions(script_path, script_perms)
+            .expect("Couldn't modify shutdown script permissions");
+
+        install_mock(&*V1, true);
+        let mut launcher = Launcher::new(None).unwrap();
+        launcher
+            .transition_state(NodeExitCode::ShouldExitLauncher)
+            .expect("Should transition state");
```

Comment:
I'm wondering if we can somehow test this without explicitly calling `transition_state`. Maybe we can create a mock of the node that exits with the code 103 (similarly to the [downgrade mock](https://github.com/casper-network/casper-node-launcher/blob/main/test_resources/downgrade.in)) and examine if launcher reacts correctly.

I didn't fully explore if that's achievable or not, just dropping it here for your consideration.

Reply:
As stated above, I'll try another mock node.

Reply:
Done.